Source/Core/Core/IOS/ES/NandUtils.cpp

```diff
@@ -393,7 +393,7 @@ std::string ESDevice::GetContentPath(const u64 title_id, const ES::Content& cont
 
 s32 ESDevice::WriteSystemFile(const std::string& path, const std::vector<u8>& data, Ticks ticks)
 {
-  auto& fs = *m_ios.GetFSDevice();
+  auto fs(std::move(*m_ios.GetFSDevice()));
```

Comment:
This doesn't look right at all. We do *not* want to move the FSDevice out of the IOS instance. (Consider what happens if WriteSystemFile is called several times, or emulated software attempts to use the FS device after WriteSystemFile has been called.)

Also, the premise of the PR seems to be flawed. There is no lifetime issue here because the FS device is guaranteed to exist for as long as the other devices exist.